### PR TITLE
Fastlane instructives for tool publishing the Gallery

### PR DESCRIPTION
**Pull request by xster**
#14305

Self contained mechanism to reduce complexity of pushing new releases to Testflight and Play Store.

The gallery package/bundle ID needed to be changed since the previous ID was already used for a google.com certificate signed app on the Play Store.

This is the first step for auto-publishing on branch cut.

### CODE REVIEW

File: examples/flutter_gallery/android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
      found in the LICENSE file.
  -->
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="io.flutter.examples.gallery">
+    package="io.flutter.demo.gallery">
```
Comment (Hixie) [inline examples/flutter_gallery/android/app/src/main/AndroidManifest.xml]: why the change from examples to demo?

**Comment by Hixie** [comment]
> The gallery package/bundle ID needed to be changed since the previous ID was already used for a google.com certificate signed app on the Play Store.

I don't understand this, can you elaborate?

File: examples/flutter_gallery/android/fastlane/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+fastlane documentation
+================
```
Comment (yjbanov) [inline examples/flutter_gallery/android/fastlane/README.md]: Could you please add one sentence explaining what we use fastlane for so that a reader does not need to google it? Something like "We use fastlane for automating building APKs and pushing then to Google Play alpha channel." Then a user who's not interested in the topic would move on immediately.

File: examples/flutter_gallery/ios/fastlane/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+fastlane documentation
```
Comment (yjbanov) [inline examples/flutter_gallery/ios/fastlane/README.md]: similar comments here

File: examples/flutter_gallery/android/fastlane/README.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+# Installation
+
+Make sure you have the latest version of the Xcode command line tools installed:
```
Comment (yjbanov) [inline examples/flutter_gallery/android/fastlane/README.md]: Is this required for Android?

**Review by yjbanov — COMMENTED** [review]
- [x] Please make sure the bundle ID change does not break devicelab tests (I can assist with that).

File: examples/flutter_gallery/android/fastlane/README.md (hunk 2 above)
Comment (xster) [reply examples/flutter_gallery/android/fastlane/README.md]: ha, I was gonna reply how frustrating it is that this file gets continuously autogen-ed, then found this https://docs.fastlane.tools/actions/skip_docs/

Comment (xster) [reply examples/flutter_gallery/android/fastlane/README.md]: Done

File: examples/flutter_gallery/android/fastlane/README.md (hunk 4 above)
Comment (xster) [reply examples/flutter_gallery/android/fastlane/README.md]: removed

File: examples/flutter_gallery/ios/fastlane/README.md (hunk 3 above)
Comment (xster) [reply examples/flutter_gallery/ios/fastlane/README.md]: done

File: examples/flutter_gallery/android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (xster) [reply examples/flutter_gallery/android/app/src/main/AndroidManifest.xml]: discussed offline

**Review by yjbanov — APPROVED** [review]

**Comment by yjbanov** [comment]
lgtm
